### PR TITLE
Remote sources

### PR DESCRIPTION
**Pull request by rpetrusha**
## Revision of loadFromRemoteSources app compat element

This PR replaces #4970 and attempts to clarify the precise effect of the config setting and the reason for setting it. 
It isn't clear to me, though, @jeffschwMSFT,when the exception is thrown -- whether it's thrown when loading the assembly with Assembly.LoadFrom, or whether it's thrown when attempting to invoke the assembly's code. It's also not clear to me which exception is thrown.

In addition, the Assembly.LoadFrom documentation should be modified to explain that executing code from remote sources is disabled by default and a link to the `loadFromRemoteSources` topic included.
  
@breyed, we'd really appreciate your feedback about the clarity of this topic. I thought that the previous version had all of the essential information, but it was so verbose and disjointed that it came close to being unintelligible.
 
Replaces #4970 

//cc @BillWagner @breyed @jeffschwMSFT @jkotas @DnlHarvey

### CODE REVIEW

**Comment by jeffschwMSFT** [comment]
Here is an example of the exception thrown (at LoadFrom time).
System.IO.FileLoadException: Could not load file or assembly 'file:assem.dll' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)
File name: 'file:assem.dll' ---> System.NotSupportedException: An attempt was made to load an assembly from a network location which would have caused the assembly to be sandboxed in previous versions of the .NET Framework. This release of the .NET Framework does not enable CAS policy by default, so this load may be dangerous. If this load is not intended to sandbox the assembly, please enable the loadFromRemoteSources switch. See http://go.microsoft.com/fwlink/?LinkId=155569 for more information.
   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks)
   at System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)

**Comment by rpetrusha** [comment]
Thanks, @jeffschwMSFT! I'll revise the topic  accordingly.

File: docs/framework/configure-apps/file-schema/runtime/loadfromremotesources-element.md [hunk 1]
```diff
@@ -25,75 +25,94 @@ Specifies whether assemblies from remote sources should be granted full trust.
...
-  
-## Example  
- The following example shows how to grant full trust to applications from remote sources.  
-  
-```xml  
+> If the assembly should not run in full trust, do not set this configuration element. Instead, create a sandboxed <xref:System.AppDomain> in which to load the assembly.
+
+The `enabled` attribute for the `<loadFromRemoteSources>` element is effective only when code access security (CAS) is disabled. By default, CAS policy is disabled in the .NET Framework 4 and later versions. If you set `enabled` to `true`, remote assemblies are granted full trust.
+
+If `enabled` is not set to `true`, a <xref:System.IO.FileLoadException> is thrown under the following conditions:
```
Comment (BillWagner) [inline docs/framework/configure-apps/file-schema/runtime/loadfromremotesources-element.md]: nit: should you specify if this means *either of the following conditions* or *both of the following conditions*?

File: docs/framework/configure-apps/file-schema/runtime/loadfromremotesources-element.md [hunk 2]
```diff
@@ -25,75 +25,94 @@ Specifies whether assemblies from remote sources should be granted full trust.
...
+
+## Notes
+
+- In the .NET Framework 4.5 and later versions, assemblies on local network shares run in full trust by default; you do not have to enable the `<loadFromRemoteSources>` element.
+
+- If an application has been copied from the web, it is flagged by Windows as being a web application, even if it resides on the local computer. You can change that designation by changing its file properties, or you can use the `<loadFromRemoteSources>` element to grant the assembly full trust. As an alternative, you can use the <xref:System.Reflection.Assembly.UnsafeLoadFrom%2A> method to load a local assembly that the operating system has flagged as having been loaded from the web.
+
+- You may get a <xref:System.IO.FileLoadException> in a Windows Virtual PC application when you try to load a file from linked folders on the hosting computer. This error may also occur when you try to load a file from a folder linked over [Remote Desktop Services](http://go.microsoft.com/fwlink/?LinkId=182775) (Terminal Services). To avoid the exception, set `enabled` to `true`.
```
Comment (BillWagner) [inline docs/framework/configure-apps/file-schema/runtime/loadfromremotesources-element.md]: This sentence reads a bit awkward to me. It's not a *Windows Virtual PC application*, but rather an *application that is running in a Windows Virtual PC*. Maybe it should be two sentences?

**Review by BillWagner — APPROVED** [review]
This is a substantial improvement, and really clarifies the behavior. 🎆 

I had one or two very small suggestions, if you want to consider them. Then, :shipit:

**Pull request merged**